src/migration.ts

```diff
@@ -49,19 +50,71 @@ export type CreateOptions = {
 
 const SEPARATOR = '_';
 
+function localeCompareStringsNumerically(a: string, b: string): number {
+  return a.localeCompare(b, undefined, {
+    usage: 'sort',
+    numeric: true,
+    sensitivity: 'variant',
+    ignorePunctuation: true,
+  });
+}
+
+function compareFileNamesByTimestamp(
+  a: string,
+  b: string,
+  logger?: Logger
+): number {
+  const aTimestamp = getTimestamp(a, logger);
+  const bTimestamp = getTimestamp(b, logger);
+
+  return aTimestamp - bTimestamp;
+}
+
+/**
+ * ! this function does not actually load the files - it only reads their names / paths from `dir`
+ * ? should the function body be adjusted to do only what can be expected from it's name
```

Comment:
**todo:** @Shinigami92 needs to checkout the code/code-path and evaluate  
_resolve open question_

Reply:
since this question is *only* about naming, it shouldn't be a showstopper

Reply:
It was more a reminder for myself when I do the deep dive upcoming weekend